databricks/koalas/tests/test_ops_on_diff_frames.py

```diff
@@ -1528,6 +1528,595 @@ def test_pow_and_rpow(self):
+
+        # other = Index
+        pandas_other = pd.Index([np.nan, 1, 3, 4, np.nan, 6], name="x")
+        koalas_other = ks.from_pandas(pandas_other)
+        koalas_other = ks.from_pandas(pandas_other)
+        self.assert_eq(pser.mod(pandas_other), kser.mod(koalas_other).sort_index())
+        self.assert_eq(pser % pandas_other, (kser % koalas_other).sort_index())
+        self.assert_eq(pser.rmod(pandas_other), kser.rmod(koalas_other).sort_index())
+
+        # other = list
+        other = [np.nan, 1, 3, 4, np.nan, 6]
+        if LooseVersion(pd.__version__) >= LooseVersion("1.2"):
+            self.assert_eq(pser.mod(other), kser.mod(other).sort_index())
+            self.assert_eq(pser % other, (kser % other).sort_index())
+            self.assert_eq(pser.rmod(other), kser.rmod(other).sort_index())
+            self.assert_eq(other % pser, (other % kser).sort_index())
+        else:
+            self.assert_eq(pser.mod(other).rename("x"), kser.mod(other).sort_index())
+            self.assert_eq((pser % other).rename("x"), (kser % other).sort_index())
+            self.assert_eq(pser.rmod(other).rename("x"), kser.rmod(other).sort_index())
+            self.assert_eq((other % pser).rename("x"), (other % kser).sort_index())
+
+        # other = tuple
+        other = (np.nan, 1, 3, 4, np.nan, 6)
+        if LooseVersion(pd.__version__) >= LooseVersion("1.2"):
+            self.assert_eq(pser.mod(other), kser.mod(other).sort_index())
+            self.assert_eq(pser % other, (kser % other).sort_index())
+            self.assert_eq(pser.rmod(other), kser.rmod(other).sort_index())
+            self.assert_eq(other % pser, (other % kser).sort_index())
+        else:
+            self.assert_eq(pser.mod(other).rename("x"), kser.mod(other).sort_index())
+            self.assert_eq((pser % other).rename("x"), (kser % other).sort_index())
+            self.assert_eq(pser.rmod(other).rename("x"), kser.rmod(other).sort_index())
+            self.assert_eq((other % pser).rename("x"), (other % kser).sort_index())
+
+        # other = list with the different length
+        other = [np.nan, 1, 3, 4, np.nan]
+        with self.assertRaisesRegex(
+            ValueError, "operands could not be broadcast together with shapes"
+        ):
+            kser.mod(other)
+        with self.assertRaisesRegex(
+            ValueError, "operands could not be broadcast together with shapes"
+        ):
+            kser % other
+        with self.assertRaisesRegex(
+            ValueError, "operands could not be broadcast together with shapes"
+        ):
+            kser.rmod(other)
+        with self.assertRaisesRegex(
+            ValueError, "operands could not be broadcast together with shapes"
+        ):
+            other % kser
+
+        # other = tuple with the different length
+        other = (np.nan, 1, 3, 4, np.nan)
+        with self.assertRaisesRegex(
+            ValueError, "operands could not be broadcast together with shapes"
```

Comment:
The error message looks weird. Is it matched with pandas'?

Reply:
The original error message from pandas looks like :

```python
ValueError: operands could not be broadcast together with shapes (4,) (8,)
```

@ueshin , maybe we don't include the `(4,) (8,)` part since it requires compute length of both objects which can be expensive ??